packages/frontend/src/features/sqlRunner/store/sqlRunnerSlice.ts

```diff
@@ -282,7 +293,11 @@ export const sqlRunnerSlice = createSlice({
                 state.successfulSqlQueries.current || '',
                 state.warehouseConnectionType,
             );
-            state.hasUnrunChanges = normalizedNewSql !== normalizedCurrentSql;
```

Comment:
I don't understand why this was not working as expected...

Reply:
see https://github.com/lightdash/lightdash/pull/12022#discussion_r1810216561